internal/api/types.go

```diff
@@ -315,30 +315,39 @@ type LanguageBackend struct {
 // a builder function which can perform this normalization and
 // validation.
 func (b *LanguageBackend) Setup() {
-	if b.Name == "" ||
-		b.Specfile == "" ||
-		b.Lockfile == "" ||
-		len(b.FilenamePatterns) == 0 ||
-		b.GetPackageDir == nil ||
-		b.Search == nil ||
-		b.Info == nil ||
-		b.Add == nil ||
-		b.Remove == nil ||
+	condition2flag := map[string]bool{
```

Comment:
could you make the reason strings a bit more friendly. At least remove the `b.`.

Reply:
yes! thank you for the reminder.

Reply:
now these are some top notch errors :ok_hand: